integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/column/TransformationInfo.java

```diff
@@ -0,0 +1,76 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.lifecycle.plan.column;
+
+import java.util.Objects;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@AllArgsConstructor
+public class TransformationInfo {
+
+  @Getter @Setter private String type;
+
+  @Getter @Setter private String subType;
+  @Getter @Setter private String description;
+  @Getter @Setter private Boolean masking;
+
+  public static TransformationInfo identity() {
+    return new TransformationInfo("DIRECT", "IDENTITY", "", false);
```

Comment:
it would work for identity as it is always non-masking and will probably have unchanging description but for `transformation` or `aggregation` which can be masking and probably could have different descriptions I would keep them as a method so they can be overloaded like `transformation(masking)` or `transformation(description, masking)`

Reply:
I would also like enums if we keep it internally in implementation - not expose them in JSON schema - as changing enums then is extremely cumbersome if someone depends on particular values.